lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -2380,13 +2381,21 @@ void PartitionCloner::visitCondBranchInst(CondBranchInst *inst) {
   auto condTy = cond->getType().getASTType();
 
   if (auto eltTy = getTensorHandleElementType(condTy)) {
-    assert(eltTy->isBuiltinIntegerType(1) && "expected Tensor<i1>");
+    // The TensorHandle's element type here could be Bool or Builtin.il.
+    // In either case, we eventually get Builtin.il to create GraphOp.
+    CanType i1CanTy;
+    if (eltTy->isBool()) {
+      i1CanTy = getSingleElementDeclFieldType(eltTy->getAnyNominal());
+    } else {
+      assert(eltTy->isBuiltinIntegerType(1) && "expected Tensor<i1>");
+      i1CanTy = eltTy->getCanonicalType();
+    }
```

Comment:
Would this be equivalent?

```
  auto &ctx = B.getASTContext();
  if (auto eltTy = getTensorHandleElementType(condTy)) {
    if (eltTy->isBool())
      eltTy = SILType::getBuiltinIntegerType(/*bitWidth*/ 1, ctx).getASTType();
  }
  assert(eltTy->isBuiltinIntegerType(1) && "expected Tensor<i1>");
```

Reply:
Just saw I suggested using `getSingleElementDeclFieldType()` on the bug. :) I don't have a preference on that vs `getBuiltinIntegerType`.

Reply:
I like the `getBuiltinIntegerType` way, since it turns out to be less code changed. Fixed in the updated code.